iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -1061,10 +1078,33 @@ private void HandleTwinResponse(MqttApplicationMessageReceivedEventArgs received
 
                     if (status != 200)
                     {
+                        IotHubClientErrorResponseMessage errorResponse = null;
+
+                        // This will only ever contain an error message which is encoded based on service contract (UTF-8).
+                        if (payloadBytes.Length > 0)
+                        {
+                            string errorResponseString = Encoding.UTF8.GetString(payloadBytes);
+                            try
+                            {
+                                errorResponse = JsonConvert.DeserializeObject<IotHubClientErrorResponseMessage>(errorResponseString);
+                            }
+                            catch (JsonException ex)
+                            {
+                                if (Logging.IsEnabled)
+                                    Logging.Error(this, $"Failed to parse twin patch error response JSON: {ex}. Message body: '{errorResponseString}'");
```

Comment:
Consider swapping the order as an exception will be many, many lines and the message body probably not nearly as long.